File: token-client-spring/src/test/java/no/nav/security/token/support/client/spring/oauth2/TestUtils.java
package no.nav.security.token.support.client.spring.oauth2;

import okhttp3.mockwebserver.MockResponse;
import okhttp3.mockwebserver.MockWebServer;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;

import java.io.IOException;
import java.util.function.Consumer;

class TestUtils {

    static void withMockServer(int port, Consumer<MockWebServer> test) throws IOException{
        MockWebServer server = new MockWebServer();
        server.start(port);
        test.accept(server);
        server.shutdown();
    }

    static void withMockServer(Consumer<MockWebServer> test) throws IOException {
        withMockServer(0, test);
    }

    static MockResponse jsonResponse(String json) {
        return new MockResponse()
            .setHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
            .setBody(json);
    }
}

File: token-client-spring/src/test/kotlin/no/nav/security/token/support/client/spring/oauth2/TestUtils.kt
package no.nav.security.token.support.client.spring.oauth2


import okhttp3.mockwebserver.MockResponse
import org.springframework.http.HttpHeaders.CONTENT_TYPE
import org.springframework.http.MediaType.APPLICATION_JSON_VALUE

internal object TestUtils {

    fun jsonResponse(json: String)=
         MockResponse().apply {
            setHeader(CONTENT_TYPE, APPLICATION_JSON_VALUE)
            setBody(json)
    }
}